library/firewall_lib.py

```diff
@@ -1468,8 +1468,8 @@ def exception_handler(exception_message):
                 changed = True
             if permanent:
                 nm_used, if_changed = try_set_zone_of_interface(module, zone, item)
-                if nm_used:
-                    changed = if_changed
+                if nm_used and if_changed:
+                    changed = True
```

Comment:
We only want the `elif` clause executed if `nm_used` is `False`.  So try this instead:

```python
                if nm_used:
                    if if_changed:
                        changed = True
```
and same below